mantidproject/mantid
# Fix run status log for aborted runs

Correct the generation of the `running` log on loading a file: if the run was aborted and then restarted, the `running` log should only show that it was running after the restart. At all times before the restart, `running` should be false. (This is important for filtering other logs by `running`, and was requested by scientists).

Add tests for old and new style logs, and also did some maintenance on the other tests in `LogParserTest`.

**To test:**

Use any datafile where the run was aborted, then restarted. For example, try `EMU00066122.nxs`, which can be found at `\\olympic\babylon5\Scratch\TomPerkins\test_tsp_filtering`:

```python
tuple = Load('EMU00066122')
ws = tuple[0]
run = ws.run()
log = run.getProperty("running")
for i in range(log.size()):
    print log.nthTime(i), log.nthValue(i)
```
Before the fix, `running` would be `True` from the start time (10:01:47) until the run was aborted at 10:02:35.
After the fix, at all times before the abort, `running` should be `False`.

Fixes #17924.

[Release notes](https://github.com/mantidproject/mantid/blob/6de56f59dafe1c8ba1aea8bcc3b2e31af2d72924/docs/source/release/v3.9.0/framework.rst#bug-fixes) have been updated.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
